bin/xbps-install/transaction.c

```diff
@@ -53,14 +53,17 @@ show_package_msgs(struct xbps_handle *xhp, xbps_object_iterator_t iter) {
 	xbps_dictionary_t obj;
 	while ((obj = xbps_object_iterator_next(iter)) != NULL) {
 		switch(xbps_transaction_pkg_type(obj)) {
-			case XBPS_TRANS_HOLD:
-				break;
 			case XBPS_TRANS_REMOVE:
-				xbps_cb_message(xhp, obj, "remove-msg");
+				xbps_cb_message(xhp, obj, "remove-msg", NULL);
 				break;
 			default:
-				xbps_cb_message(xhp, obj, "install-msg");
-				break;
+				{
+					const char *pkgname = xbps_string_cstring_nocopy(xbps_dictionary_get(obj, "pkgname"));
+					xbps_dictionary_t pkgdb_pkg = pkgname ? xbps_pkgdb_get_pkg(xhp, pkgname) : NULL;
+					xbps_object_t previous = pkgdb_pkg ? xbps_dictionary_get(pkgdb_pkg, "install-msg") : NULL;
+					xbps_cb_message(xhp, obj, "install-msg", previous);
+					break;
```

Comment:
Can you put a comment explaining this? I don't get the logic.

Reply:
Here install-msg of installed version of package is read from system pkgdb, and passed to xbps_cb_message.
Then xbps_cb_message don't print message if current and previous are same.
Is doc comment of xbps_cb_message best place to write it down?

Reply:
>  Then xbps_cb_message don't print message if current and previous are same.

Not sure how I feel about this... If someone misses an INSTALL.msg, they will never see it again.

>  Is doc comment of xbps_cb_message best place to write it down?

Yeah, I guess so. And comment this code block "get installed version of package, if any".